extensions/amp-experiment/0.1/amp-experiment.js

```diff
@@ -36,7 +38,49 @@ export class AmpExperiment extends AMP.BaseElement {
       toggle(this.element, false);
       return;
     }
-    dev.info(EXPERIMENT, 'amp-experiment is on');
+
+    const config = this.getConfig_();
+    const results = {};
+    this.experimentVariants = Promise.all(
+        Object.keys(config).map(experimentName => {
+          return this.getVariantAllocation_(config[experimentName])
+              .then(variantName => {
+                if (variantName) {
+                  results[experimentName] = variantName;
+                }
+              });
+        })).then(() => results);
+    this.experimentVariants.then(this.addToBody_());
+  }
+
+  getConfig_() {
+    const children = this.element.children;
+    user.assert(
+        children.length == 1 && children[0].tagName.toUpperCase() == 'SCRIPT'
+            && children[0].getAttribute('type').toUpperCase()
+                == 'APPLICATION/JSON',
+        '<amp-experiment> should contain exactly one ' +
+        '<script type="application/json"> child.');
+
+    return JSON.parse(children[0].textContent);
```

Comment:
I think we actually want things to break here, instead of being hidden from user. It is a user config error should be fixed before getting published.